src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -2474,7 +2484,13 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
           if (result0.nonEmpty) checkPure(result0, supple = true)
         }
 
-        treeCopy.Block(block, statsTyped, expr1)
+        // Remove ValDef for right-associative by-value operator desugaring which has been inlined into expr1
+        val statsTyped2 = statsTyped match {
+          case (vd: ValDef) :: Nil if inlinedRightAssocValDefs contains vd.symbol => Nil
```

Comment:
Could eagerly clear the entry from the Map here `if (inlinedRightAssocValDefs.remove(vd.symbol).isDefined) =>`